jsonschema_lexer/_lexer.py

```diff
@@ -24,109 +27,162 @@ class JSONSchemaLexer(JsonLexer):
     ]
 
     data_types: ClassVar[list[str]] = [
-        "object",
-        "integer",
-        "string",
-        "number",
-        "array",
-        "boolean",
-        "null",
-    ]
-    core_keywords: ClassVar[list[str]] = [
-        "$schema",
-        "$id",
-        "$ref",
-        "$defs",
-        "$comment",
-        "$dynamicAnchor",
-        "$dynamicRef",
-        "$anchor",
-        "$vocabulary",
-    ]
-    applicator_keywords: ClassVar[list[str]] = [
-        "oneOf",
-        "allOf",
-        "anyOf",
-        "if",
-        "then",
-        "else",
-        "not",
-        "properties",
-        "patternProperties",
-        "additionalProperties",
-        "dependentSchemas",
-        "propertyNames",
-        "prefixItems",
-        "contains",
-        "items",
-    ]
-    meta_data_keywords: ClassVar[list[str]] = [
-        "title",
-        "description",
-        "default",
-        "deprecated",
-        "examples",
-        "readOnly",
-        "writeOnly",
-    ]
-    validation_keywords: ClassVar[list[str]] = [
-        "type",
-        "enum",
-        "const",
-        "minLength",
-        "maxLength",
-        "pattern",
-        "maximum",
-        "exclusiveMinimum",
-        "multipleOf",
-        "exclusiveMaximum",
-        "minimum",
-        "dependentRequired",
-        "minProperties",
-        "maxProperties",
-        "required",
-        "minItems",
-        "maxItems",
-        "minContains",
-        "maxContains",
-        "uniqueItems",
-    ]
-    other_keywords: ClassVar[list[str]] = [
-        "format",
-        "unevaluatedItems",
-        "unevaluatedProperties",
-        "contentEncoding",
-        "contentMediaType",
-        "contentSchema",
-        "format_assertion",
+        '"object"',
+        '"integer"',
+        '"string"',
+        '"number"',
+        '"array"',
+        '"boolean"',
+        '"null"',
     ]
+    keywords: ClassVar[dict[str | None, list[str]]] = {}
+    identifier: ClassVar[dict[str | None, str]] = {}
+    default_dialect = None
+
+    def __init__(self, default_dialect: str | None = None):
+        super().__init__()  # type: ignore[reportUnknownMemberType]
+        self._populate_keywords_and_identifiers()
+        if default_dialect and default_dialect[0] != '"':
+            default_dialect = '"' + default_dialect
+
+        if default_dialect and default_dialect[-1] != '"':
+            default_dialect = default_dialect + '"'
+
+        self.default_dialect = default_dialect
+
+    def _populate_keywords_and_identifiers(self):
+        dialect_files = files("jsonschema_lexer") / "data" / "keywords"
+        if not dialect_files.is_dir():
```

Comment:
This is copied from Bowtie I guess, but here your data is always inside the package, so I think if my mental compiler works this path is wrong. Also you load all the dialects, but in theory you could only load the ones for the dialect you need, no?

Reply:
I am loading all the dialects as when the lexer proceeds further, it might be that it encounters a new schema specified. Instead of this I could also load dialects as required if you say.

Reply:
Umm I am not sure why the path would be wrong though?

Reply:
Because `Path(__file__).parent.parent` is the root directory of the repository, so that resolves to `/data/keywords` rather than `jsonschema_lexer/data/keywords`.

Reply:
Oh yeah you are right, `dialect_files = files("jsonschema_lexer") / "data" / "keywords"` itself worked while testing so it never went inside that condition so missed it.